siad/apiwallet.go

```diff
@@ -15,8 +15,9 @@ func (d *daemon) walletAddressHandler(w http.ResponseWriter, req *http.Request)
 		return
 	}
 
-	// TODO: Explain what's happening here, and why it doesn't look like all of
-	// the other calls.
+	// Since coinAddress is not a struct, we define one here so that writeJSON
+	// writes an object instead of a bare value. In addition, we transmit the
+	// coinAddress as a hex-encoded string rather than a byte array.
 	writeJSON(w, struct {
 		Address string
 	}{fmt.Sprintf("%x", coinAddress)})
```

Comment:
hmm. I never really thought about it, but you have a point. It looks like json will marshal a byte array as a json array, so it wouldn't be too difficult to work with. I don't know how easy it is to convert a json array to a hex-encoded string though.